DCOS/dcos-testing.sh

```diff
@@ -904,6 +905,121 @@ test_windows_agent_graceful_shutdown() {
+    eval "cat <<-EOF
+	$(cat $WINDOWS_APP_CONTAINER_TEMPLATE | jq -r 'del(.constraints[1])')
+	EOF
+	" > $WINDOWS_APP_CONTAINER_RENDERED_TEMPLATE
+
+    echo "Deploying a Windows Marathon application on DC/OS"
+
+    dcos marathon app add $WINDOWS_APP_CONTAINER_RENDERED_TEMPLATE || {
+        echo "ERROR: Failed to deploy the Windows Marathon application"
+        return 1
+    }
+    
+    local APP_NAME=$(get_marathon_application_name $WINDOWS_APP_CONTAINER_RENDERED_TEMPLATE)
+    
+    $DIR/utils/check-marathon-app-health.py --name $APP_NAME || {
+        echo "ERROR: Failed to get $APP_NAME application health checks"
+        dcos marathon app show $APP_NAME > "${TEMP_LOGS_DIR}/dcos-marathon-${APP_NAME}-app-details.json"
+        return 1
+    }
+    local PORT=$(get_marathon_application_host_port $WINDOWS_APP_CONTAINER_RENDERED_TEMPLATE)
+    local AGENT_HOSTNAME=$(dcos marathon app show $APP_NAME | jq -r ".tasks[0].host")
+    test_dcos_task_connectivity "$APP_NAME" "$AGENT_HOSTNAME" "$AGENT_ROLE" "$PORT" || return 1
+    
+    #
+    #### Killing the service
+    #
+    setup_remote_winrm_client || return 1
+    upload_files_via_scp -i $PRIVATE_SSH_KEY_PATH -u $LINUX_ADMIN -h $MASTER_PUBLIC_ADDRESS -p "2200" -f "/tmp/kill-mesos-agent-ungraceful.ps1" "$DIR/utils/kill-mesos-agent-ungraceful.ps1" || {
+        echo "ERROR: Failed to scp kill-mesos-agent-ungraceful.ps1"
+        return 1
+    }
+    echo "Killing the dcos-mesos-slave service on $AGENT_HOSTNAME and disabling recovery options"
+    local REMOTE_CMD_STOP="/tmp/wsmancmd -H $AGENT_HOSTNAME -s -a basic -u $WIN_AGENT_ADMIN -p $WIN_AGENT_ADMIN_PASSWORD --powershell --file /tmp/kill-mesos-agent-ungraceful.ps1"
+    local CHECK_RESULT_STOP=$(run_ssh_command -i $PRIVATE_SSH_KEY_PATH -u $LINUX_ADMIN -h $MASTER_PUBLIC_ADDRESS -p "2200" -c "$REMOTE_CMD_STOP" 2>/dev/null) || {
+        echo -e "Error: Failed to run service kill script on $AGENT_HOSTNAME"
+        return 1
+    }
+
+    if [ "$CHECK_RESULT_STOP" == "SUCCESS" ]; then
+        echo "Service dcos-mesos-slave successfully Stopped!"
+    elif [ "$CHECK_RESULT_STOP" == "FAILURE" ]; then
+        echo "ERROR: Service dcos-mesos-slave stopping failure"
+        return 1
+    else
+        echo "ERROR: Service dcos-mesos-slave is in '$CHECK_RESULT' state, which is unknown"
+        return 1
+    fi
+
+    echo "Waiting with a timeout of 5mins for DCOS to fail the task over from $AGENT_HOSTNAME..."
+    local NEW_TASK_HOST=""
+    SECONDS=0
+    while true; do
+        if [[ $SECONDS -gt 300 ]]; then
+            echo "ERROR: task for $APP_NAME didn't migrate from $AGENT_HOSTNAME within $TIMEOUT seconds"
+            return 1
+        fi
+        NEW_TASK_HOST=$(dcos marathon app show $APP_NAME | jq -r ".tasks[0].host")
+        if [[ $NEW_TASK_HOST != $AGENT_HOSTNAME ]]; then
+            echo "Task successfully fail-overed from $AGENT_HOSTNAME to $NEW_TASK_HOST"    
+            break
+        else
```

Comment:
You don't really need an `else` here, since you exit the loop in case `then` is satisfied.